src/openvslam/system.cc

```diff
@@ -246,7 +246,8 @@ Mat44_t system::feed_monocular_frame(const cv::Mat& img, const double timestamp,
     const Mat44_t cam_pose_cw = tracker_->track_monocular_image(img, timestamp, mask);
 
     frame_publisher_->update(tracker_);
-    if (tracker_->tracking_state_ == tracker_state_t::Tracking) {
+    if (tracker_->tracking_state_ == tracker_state_t::Tracking ||
+        tracker_->finish_update_pose_request()) {
```

Comment:
It looks unnecessary. Or I think there is some misunderstanding.

Reply:
This is the place why the `finish_update_pose_request()` routine was created. If the requested pose was intentionally set to specific one the returned back by `track_*_frame()` pose also will be set to a given one even if the relocalization-by-pose was failed. However, in the failed case if pose was requested to set, we should update the pose in `map_publisher` which will be seen on the screen e.g. when using Pangolin viewer. In other words, we need to distinguish between the situation when relocalization was really failed (and thus we keep the last correct pose) and the situation when someone intentionally set a pose but relocalization was failed. In the second case it need to display the pose the way it was set by request.

Another option to make it: is to introduce a new state, say `tracker_state_t::PoseSet` or something else. However, I do not think, that tracker' state deserves such a big changes for only one use-case. So, there were `finish_update_pose_request()` routine introduced to return back an information about did pose request was made or not (and to reset pose request indicator if set).

Reply:
I realized that I was using a short evaluation. It's not intuitive, so please clearly separate flag changes from flag checks.

Reply:
After successful relocalization, it would be better to set update_pose_is_requested_ to false in the tracking_module.
It is enough for system to observe tracking_state_.

Reply:
> In other words, we need to distinguish between the situation when relocalization was really failed (and thus we keep the last correct pose) and the situation when someone intentionally set a pose but relocalization was failed. In the second case it need to display the pose the way it was set by request.

I think it is enough to call map_publisher_->set_current_cam_pose in system::update_pose. (and map_publisher_->set_current_cam_pose_wc)

Reply:
> I think it is enough to call map_publisher_->set_current_cam_pose in system::update_pose. (and map_publisher_->set_current_cam_pose_wc)

Good! Update preliminarily the `map_publisher_` before `track_*` method call will work. [Applied](https://github.com/OpenVSLAM-Community/openvslam/pull/79/files#diff-0bc37256b42a468d11db219725a5b24ef9478f88143cb2c5e939634cf260520fR291-R294) in the latest commit.